android/modules/analytics/src/java/ti/modules/titanium/analytics/AnalyticsModule.java

```diff
@@ -86,23 +87,38 @@ public void filterEvents(Object eventsObj) {
 	}
 
 	@Kroll.method
-	public void featureEvent(String event, @Kroll.argument(optional = true) KrollDict data)
+	public int featureEvent(String event, @Kroll.argument(optional = true) KrollDict data)
 	{
-		if (TiApplication.getInstance().isAnalyticsEnabled()) {
-			if (data instanceof HashMap) {
-				analytics.sendAppFeatureEvent(event, TiConvert.toJSON(data));
-			} else if (data != null) {
-				try {
-					analytics.sendAppFeatureEvent(event, new JSONObject(data.toString()));
-				} catch (JSONException e) {
-					Log.e(TAG, "Cannot convert data into JSON");
-				}
-			} else {
-				analytics.sendAppFeatureEvent(event, null);
-			}
-		} else {
-			Log.e(TAG, "Analytics is disabled.  To enable, please update the <analytics></analytics> node in your tiapp.xml");
-		}
+	    if (TiApplication.getInstance().isAnalyticsEnabled()) {
+	        if (data instanceof HashMap) {
+	            JSONObject jsonData = TiConvert.toJSON(data);
+	            if (TiConvert.validateJSON(jsonData, 0) == TiC.ERROR_CODE_NO_ERROR) {
+	                analytics.sendAppFeatureEvent(event, jsonData);
+	                return TiC.ERROR_CODE_NO_ERROR;
+	            } else {
+	                return TiC.ERROR_CODE_UNKNOWN;
+	            }
+	        } else if (data != null) {
+	            try {
+	                JSONObject jsonData = new JSONObject(data.toString());
+	                if (TiConvert.validateJSON(jsonData, 0) == TiC.ERROR_CODE_NO_ERROR) {
+	                    analytics.sendAppFeatureEvent(event, jsonData);
+	                    return TiC.ERROR_CODE_NO_ERROR;
+	                } else {
+	                    return TiC.ERROR_CODE_UNKNOWN;
```

Comment:
`"Feature event <event> not conforming to recommended usage."`
Good to have : Try to log the event in question. This can help developer since app emits lot of events.